test/common/scale_up_replicas_threescale.go

```diff
@@ -0,0 +1,152 @@
+package common
+
+import (
+	goctx "context"
+	"fmt"
+	"testing"
+	"time"
+
+	threescalev1 "github.com/3scale/3scale-operator/pkg/apis/apps/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	numberOfReplicas int64 = 2
+	scaleUpReplicas  int64 = 3
+	name                   = "3scale"
+	namespace              = "redhat-rhmi-3scale"
+)
+
+func TestReplicasInThreescale(t *testing.T, ctx *TestingContext) {
+
+	createWorkspaceRetryInterval := time.Second * 20
+	createWorkspaceTimeout := time.Minute * 7
+
+	apim, err := getAPIManager(ctx.Client, t)
+	if err != nil {
+		t.Fatalf("failed to get APIManager : %v", err)
+	}
+
+	t.Log("Checking correct number of replicas are set")
+	if err := checkNumberOfReplicasAgainstValue(apim, numberOfReplicas, createWorkspaceRetryInterval, createWorkspaceTimeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas to start : %v", err)
+	}
+
+	apim, err = updateAPIManager(ctx.Client, t, scaleUpReplicas)
+	if err != nil {
+		t.Fatalf("Unable to update : %v", err)
+	}
+
+	t.Log("Checking correct number of updated replicas are set")
+	if err := checkNumberOfReplicasAgainstValue(apim, scaleUpReplicas, createWorkspaceRetryInterval, createWorkspaceTimeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+
+	apim, err = updateAPIManager(ctx.Client, t, numberOfReplicas)
+	if err != nil {
+		t.Fatalf("Unable to update : %v", err)
+	}
+
+	t.Log("Checking correct number of replicas has been reset")
+	if err := checkNumberOfReplicasAgainstValue(apim, numberOfReplicas, createWorkspaceRetryInterval, createWorkspaceTimeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+}
+
+func getAPIManager(dynClient k8sclient.Client, t *testing.T) (threescalev1.APIManager, error) {
+	apim := &threescalev1.APIManager{}
+
+	if err := dynClient.Get(goctx.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, apim); err != nil {
+		t.Fatalf("Error : %v ", err)
```

Comment:
```suggestion
		return nil, err
```
Would be good to return the error here. You are error handling anyway in the main function. You can avoid passing in `t` into the function.